File: backend/src/main/java/com/uniform/store/service/CartService.java
package com.uniform.store.service;

import com.uniform.store.dto.request.AddCartItemRequest;
import com.uniform.store.dto.request.MergeCartRequest;
import com.uniform.store.dto.request.UpdateCartItemRequest;
import com.uniform.store.dto.response.CartDto;

public interface CartService {

    CartDto getCart(String email);

    CartDto addItem(String email, AddCartItemRequest req);

    CartDto updateItem(String email, Long itemId, UpdateCartItemRequest req);

    CartDto removeItem(String email, Long itemId);

    CartDto clearCart(String email);

    CartDto mergeCart(String email, MergeCartRequest req);
}

File: frontend/src/services/cartService.js
import apiClient from '../lib/apiClient';

function unwrap(response) {
  const body = response?.data;
  if (body && body.success === false) {
    throw new Error(body.message || 'Request failed');
  }
  return body?.data;
}

export async function getCart() {
  const resp = await apiClient.get('/cart');
  return unwrap(resp);
}

export async function addItem({ variantId, quantity }) {
  const resp = await apiClient.post('/cart/items', { variantId, quantity });
  return unwrap(resp);
}

export async function updateItem(itemId, quantity) {
  const resp = await apiClient.patch(`/cart/items/${itemId}`, { quantity });
  return unwrap(resp);
}

export async function removeItem(itemId) {
  const resp = await apiClient.delete(`/cart/items/${itemId}`);
  return unwrap(resp);
}

export async function clearCart() {
  const resp = await apiClient.delete('/cart');
  return unwrap(resp);
}

export async function mergeCart(items) {
  const resp = await apiClient.post('/cart/merge', { items });
  return unwrap(resp);
}
